diff -Naur Mesa-8.0.3.orig/src/gallium/auxiliary/util/u_debug_symbol.c Mesa-8.0.3/src/gallium/auxiliary/util/u_debug_symbol.c
--- Mesa-8.0.3.orig/src/gallium/auxiliary/util/u_debug_symbol.c	2011-12-07 23:54:22.000000000 +0000
+++ Mesa-8.0.3/src/gallium/auxiliary/util/u_debug_symbol.c	2012-07-01 22:35:05.272342392 +0000
@@ -151,7 +151,7 @@
 }
 #endif
 
-#ifdef __GLIBC__
+#if defined(__GLIBC__)  && !defined(__UCLIBC__)
 #include <execinfo.h>
 
 /* This can only provide dynamic symbols, or binary offsets into a file.
diff -Naur Mesa-8.0.3.orig/src/glsl/strtod.c Mesa-8.0.3/src/glsl/strtod.c
--- Mesa-8.0.3.orig/src/glsl/strtod.c	2012-05-18 04:27:14.000000000 +0000
+++ Mesa-8.0.3/src/glsl/strtod.c	2012-07-01 18:19:20.624711636 +0000
@@ -45,7 +45,7 @@
 glsl_strtod(const char *s, char **end)
 {
 #if defined(_GNU_SOURCE) && !defined(__CYGWIN__) && !defined(__FreeBSD__) && \
-   !defined(__HAIKU__)
+   !defined(__HAIKU__) && !defined(__UCLIBC__)
    static locale_t loc = NULL;
    if (!loc) {
       loc = newlocale(LC_CTYPE_MASK, "C", NULL);
diff -Naur Mesa-8.0.3.orig/src/mesa/main/imports.c Mesa-8.0.3/src/mesa/main/imports.c
--- Mesa-8.0.3.orig/src/mesa/main/imports.c	2012-05-18 04:27:14.000000000 +0000
+++ Mesa-8.0.3/src/mesa/main/imports.c	2012-07-01 20:36:48.235060134 +0000
@@ -767,7 +767,7 @@
 _mesa_strtof( const char *s, char **end )
 {
 #if defined(_GNU_SOURCE) && !defined(__CYGWIN__) && !defined(__FreeBSD__) && \
-   !defined(ANDROID) && !defined(__HAIKU__)
+   !defined(ANDROID) && !defined(__HAIKU__) && !defined(__UCLIBC__)
    static locale_t loc = NULL;
    if (!loc) {
       loc = newlocale(LC_CTYPE_MASK, "C", NULL);
