diff -Naur Mesa-7.11.2.orig/configure Mesa-7.11.2/configure
--- Mesa-7.11.2.orig/configure	2012-01-06 12:53:32.000000000 -0500
+++ Mesa-7.11.2/configure	2012-01-06 12:54:20.000000000 -0500
@@ -11401,6 +11401,8 @@
 echo "        Shared libs:     $enable_shared"
 echo "        Static libs:     $enable_static"
 
+DEFINES+=" -U__GLIBC__ -U_GNU_SOURCE"
+
 # cleanup the CFLAGS/CXXFLAGS/DEFINES vars
 cflags=`echo $CFLAGS $OPT_FLAGS $PIC_FLAGS $ARCH_FLAGS | \
     $SED 's/^ *//;s/  */ /;s/ *$//'`
