diff -Naur libtirpc-0.2.2.orig//src/Makefile.am libtirpc-0.2.2/src/Makefile.am
--- libtirpc-0.2.2.orig//src/Makefile.am	2011-05-02 08:10:40.000000000 -0400
+++ libtirpc-0.2.2/src/Makefile.am	2012-02-23 21:32:57.000000000 -0500
@@ -63,7 +63,6 @@
     libtirpc_la_CFLAGS = -DHAVE_RPCSEC_GSS $(GSSGLUE_CFLAGS)
 endif
 
-## libtirpc_a_SOURCES += key_call.c key_prot_xdr.c getpublickey.c
-## libtirpc_a_SOURCES += netname.c netnamer.c rpcdname.c \
-## libtirpc_a_SOURCES += rtime.c \
-## auth_time.c auth_des.c authdes_prot.c 
+libtirpc_la_SOURCES += key_prot_xdr.c getpublickey.c
+libtirpc_la_SOURCES += netname.c netnamer.c rpcdname.c
+libtirpc_la_SOURCES += rtime.c
