diff -ruN /var/tmp/portage/sys-boot/grub-0.97-r6/work/grub-0.97/configure.ac /root/hardened/grub-0.97/configure.ac
--- /work/grub-0.97/configure.ac	2008-06-03 16:09:34.000000000 +0200
+++ /grub-0.97/configure.ac	2008-06-04 00:24:01.000000000 +0200
@@ -106,6 +106,7 @@
     ])
     if test "x$no_stack_protector_flag" = xyes; then
       STAGE2_CFLAGS="$STAGE2_CFLAGS -fno-stack-protector"
+      GRUB_CFLAGS="$GRUB_CFLAGS -fno-stack-protector"
     fi
   fi
 fi
