--- gcc/Makefile.in	2009-03-10 00:13:55.000000000 +0000
+++ gcc/Makefile.in	2009-03-10 00:19:02.000000000 +0000
@@ -623,7 +623,7 @@
 # This is the variable actually used when we compile.
 # If you change this line, you probably also need to change the definition
 # of HOST_CFLAGS in build-make to match.
-ALL_CFLAGS = $(X_CFLAGS) $(T_CFLAGS) \
+ALL_CFLAGS = $(X_CFLAGS) $(T_CFLAGS) -fno-stack-protector \
   $(CFLAGS) $(INTERNAL_CFLAGS) $(COVERAGE_FLAGS) $(WARN_CFLAGS) $(XCFLAGS) @DEFS@
 
 # Likewise.
