diff -Nru /var/tmp/portage/sys-fs/e2fsprogs-1.40.11-r1/work/e2fsprogs-1.40.11/configure.in /root/hardened/e2fsprogs-1.40.11/configure.in
--- /configure.in	2008-07-04 01:34:23.000000000 +0200
+++ /configure.in	2008-07-04 01:29:56.000000000 +0200
@@ -219,10 +219,12 @@
 then
 	ELF_CMT=#
 	MAKEFILE_ELF=/dev/null
+	ELF_CFLAGS=""
 	echo "Disabling ELF shared libraries"
 else
 	E2_PKG_CONFIG_STATIC=
 	ELF_CMT=
+	ELF_CFLAGS="-fPIC"
 	MAKEFILE_ELF=$srcdir/lib/Makefile.elf-lib
 	[case "$host_os" in
 	solaris2.*)
@@ -235,6 +237,7 @@
 fi
 ,
 MAKEFILE_ELF=/dev/null
+ELF_CFLAGS=""
 ELF_CMT=#
 echo "Disabling ELF shared libraries by default"
 )
diff -Nru /var/tmp/portage/sys-fs/e2fsprogs-1.40.11-r1/work/e2fsprogs-1.40.11/lib/uuid/Makefile.in /root/hardened/e2fsprogs-1.40.11/lib/uuid/Makefile.in
--- /lib/uuid/Makefile.in	2008-07-04 01:34:22.000000000 +0200
+++ /lib/uuid/Makefile.in	2008-07-04 01:16:55.000000000 +0200
@@ -61,10 +61,10 @@
 
 .c.o:
 	@echo "	CC $<"
-	@$(CC) $(ALL_CFLAGS) -c $< -o $@
+	@$(CC) $(ALL_CFLAGS) $(ELF_CFLAGS) -c $< -o $@
 @PROFILE_CMT@	@$(CC) $(ALL_CFLAGS) -g -pg -o profiled/$*.o -c $<
 @CHECKER_CMT@	@$(CC) $(ALL_CFLAGS) -checker -g -o checker/$*.o -c $<
-@ELF_CMT@	@$(CC) $(ALL_CFLAGS) -fPIC -o elfshared/$*.o -c $<
+@ELF_CMT@	@$(CC) $(ALL_CFLAGS) $(ELF_CFLAGS) -o elfshared/$*.o -c $<
 @BSDLIB_CMT@	@$(CC) $(ALL_CFLAGS) $(BSDLIB_PIC_FLAG) -o pic/$*.o -c $<
 
 all:: tst_uuid uuid_time $(SMANPAGES) uuid.pc
diff -Nru /var/tmp/portage/sys-fs/e2fsprogs-1.40.11-r1/work/e2fsprogs-1.40.11/MCONFIG.in /root/hardened/e2fsprogs-1.40.11/MCONFIG.in
--- /MCONFIG.in	2008-07-04 01:34:24.000000000 +0200
+++ /MCONFIG.in	2008-07-04 01:29:30.000000000 +0200
@@ -36,6 +36,7 @@
 BUILD_CC = @BUILD_CC@
 DEFS = -DLOCALEDIR=\"$(localedir)\" -DROOT_SYSCONFDIR=\"$(root_sysconfdir)\" @DEFS@
 CFLAGS = @CFLAGS@
+ELF_CFLAGS = -fPIC
 CPPFLAGS =  -I$(top_builddir)/lib -I$(top_srcdir)/lib @CPPFLAGS@
 INTL_FLAGS = @INTL_FLAGS@
 ALL_CFLAGS = $(CPPFLAGS) $(DEFS) $(USE_WFLAGS) $(CFLAGS) $(XTRA_CFLAGS) \
