diff -Nuar openssh-6.4p1.orig/configure.ac openssh-6.4p1/configure.ac
--- openssh-6.4p1.orig/configure.ac	2014-01-22 17:05:13.463582250 +0000
+++ openssh-6.4p1/configure.ac	2014-01-22 17:13:08.102608790 +0000
@@ -223,7 +223,7 @@
     [AC_LANG_PROGRAM([[
 #include <stdlib.h>
 __attribute__((__unused__)) static void foo(void){return;}]],
-    [[ exit(0); ]])],
+    [[ return 0; ]])],
     [ AC_MSG_RESULT([yes]) ],
     [ AC_MSG_RESULT([no])
       AC_DEFINE(NO_ATTRIBUTE_ON_RETURN_TYPE, 1,
@@ -410,7 +410,7 @@
 	    [AC_LANG_PROGRAM([[
 #define testmacro foo
 #define testmacro bar]],
-	    [[ exit(0); ]])],
+	    [[ return 0; ]])],
 	    [ AC_MSG_RESULT([yes]) ],
 	    [ AC_MSG_RESULT([no])
 	      CC="`echo $CC | sed 's/-qlanglvl\=ansi//g'`"
@@ -526,9 +526,9 @@
 	AC_MSG_CHECKING([if we have working getaddrinfo])
 	AC_RUN_IFELSE([AC_LANG_SOURCE([[ #include <mach-o/dyld.h>
 main() { if (NSVersionOfRunTimeLibrary("System") >= (60 << 16))
-		exit(0);
+		return 0;
 	else
-		exit(1);
+		return 1;
 }
 			]])],
 	[AC_MSG_RESULT([working])],
@@ -1016,7 +1016,7 @@
 esac
 
 AC_MSG_CHECKING([compiler and flags for sanity])
-AC_RUN_IFELSE([AC_LANG_PROGRAM([[ #include <stdio.h> ]], [[ exit(0); ]])],
+AC_RUN_IFELSE([AC_LANG_PROGRAM([[ #include <stdio.h> ]], [[ return 0; ]])],
 	[	AC_MSG_RESULT([yes]) ],
 	[
 		AC_MSG_RESULT([no])
@@ -1048,9 +1048,9 @@
     strncpy(buf,"/etc", 32);
     s = dirname(buf);
     if (!s || strncmp(s, "/", 32) != 0) {
-	exit(1);
+	return 1;
     } else {
-	exit(0);
+	return 0;
     }
 }
 				]])],
@@ -1140,19 +1140,19 @@
 	int a=0, b=0, c=0, d=0, n, v;
 	n = sscanf(ZLIB_VERSION, "%d.%d.%d.%d", &a, &b, &c, &d);
 	if (n != 3 && n != 4)
-		exit(1);
+		return 1;
 	v = a*1000000 + b*10000 + c*100 + d;
 	fprintf(stderr, "found zlib version %s (%d)\n", ZLIB_VERSION, v);
 
 	/* 1.1.4 is OK */
 	if (a == 1 && b == 1 && c >= 4)
-		exit(0);
+		return 0;
 
 	/* 1.2.3 and up are OK */
 	if (v >= 1020300)
-		exit(0);
+		return 0;
 
-	exit(2);
+	return 2;
 	]])],
 	AC_MSG_RESULT([no]),
 	[ AC_MSG_RESULT([yes])
@@ -1254,7 +1254,7 @@
 #include <dirent.h>]],
 	[[
 	struct dirent d;
-	exit(sizeof(d.d_name)<=sizeof(char));
+	return (sizeof(d.d_name)<=sizeof(char));
 	]])],
 	[AC_MSG_RESULT([yes])],
 	[
@@ -1300,7 +1300,7 @@
 #include <skey.h>
 				]], [[
 	char *ff = skey_keyinfo(""); ff="";
-	exit(0);
+	return 0;
 				]])],
 				[AC_MSG_RESULT([yes])],
 				[
@@ -1405,7 +1405,7 @@
 #include <stdlib.h>
 #include <stdint.h>
 #include <ldns/ldns.h>
-int main() { ldns_status status = ldns_verify_trusted(NULL, NULL, NULL, NULL); status=LDNS_STATUS_OK; exit(0); }
+int main() { ldns_status status = ldns_verify_trusted(NULL, NULL, NULL, NULL); status=LDNS_STATUS_OK; return 0; }
                                 ]])
                 ],
 				[AC_MSG_RESULT(yes)],
@@ -1462,7 +1462,7 @@
 		    [[
 	int i = H_SETSIZE;
 	el_init("", NULL, NULL, NULL);
-	exit(0);
+	return 0;
 		    ]])],
 		    [ AC_MSG_RESULT([yes]) ],
 		    [ AC_MSG_RESULT([no])
@@ -1757,9 +1757,9 @@
 	errno=0;
 	setresuid(0,0,0);
 	if (errno==ENOSYS)
-		exit(1);
+		return 1;
 	else
-		exit(0);
+		return 0;
 		]])],
 		[AC_MSG_RESULT([yes])],
 		[AC_DEFINE([BROKEN_SETRESUID], [1],
@@ -1780,9 +1780,9 @@
 	errno=0;
 	setresgid(0,0,0);
 	if (errno==ENOSYS)
-		exit(1);
+		return 1;
 	else
-		exit(0);
+		return 0;
 		]])],
 		[AC_MSG_RESULT([yes])],
 		[AC_DEFINE([BROKEN_SETRESGID], [1],
@@ -1824,7 +1824,7 @@
 		[[
 	char b[5];
 	snprintf(b,5,"123456789");
-	exit(b[4]!='\0'); 
+	return (b[4]!='\0'); 
 		]])],
 		[AC_MSG_RESULT([yes])],
 		[
@@ -1857,7 +1857,7 @@
 }
 		]], [[
 	char x[1];
-	exit(x_snprintf(x, 1, "%s %d", "hello", 12345) == 11 ? 0 : 1);
+	return (x_snprintf(x, 1, "%s %d", "hello", 12345) == 11 ? 0 : 1);
 		]])],
 		[AC_MSG_RESULT([yes])],
 		[
@@ -1909,9 +1909,9 @@
 	]], [[
 	char template[]="conftest.mkstemp-test";
 	if (mkstemp(template) == -1)
-		exit(1);
+		return 1;
 	unlink(template);
-	exit(0);
+	return 0;
 	]])],
 	[
 		AC_MSG_RESULT([no])
@@ -1942,22 +1942,22 @@
 
 	pid = fork();
 	if (pid < 0) {		/* failed */
-		exit(1);
+		return 1;
 	} else if (pid > 0) {	/* parent */
 		waitpid(pid, &status, 0);
 		if (WIFEXITED(status))
-			exit(WEXITSTATUS(status));
+			return (WEXITSTATUS(status));
 		else
-			exit(2);
+			return 2;
 	} else {		/* child */
 		close(0); close(1); close(2);
 		setsid();
 		openpty(&ptyfd, &ttyfd, NULL, NULL, NULL);
 		fd = open("/dev/tty", O_RDWR | O_NOCTTY);
 		if (fd >= 0)
-			exit(3);	/* Acquired ctty: broken */
+			return 3;	/* Acquired ctty: broken */
 		else
-			exit(0);	/* Did not acquire ctty: OK */
+			return 0;	/* Did not acquire ctty: OK */
 	}
 		]])],
 		[
@@ -1998,7 +1998,7 @@
 	err = getaddrinfo(name, TEST_PORT, &hints, &gai_ai);
 	if (err != 0) {
 		fprintf(stderr, "getaddrinfo failed (%s)", gai_strerror(err));
-		exit(1);
+		return 1;
 	}
 
 	for (ai = gai_ai; ai != NULL; ai = ai->ai_next) {
@@ -2015,7 +2015,7 @@
 			else
 				fprintf(stderr, "getnameinfo failed: %s\n",
 				    gai_strerror(err));
-			exit(2);
+			return 2;
 		}
 
 		sock = socket(ai->ai_family, ai->ai_socktype, ai->ai_protocol);
@@ -2023,10 +2023,10 @@
 			perror("socket");
 		if (bind(sock, ai->ai_addr, ai->ai_addrlen) < 0) {
 			if (errno == EBADF)
-				exit(3);
+				return 3;
 		}
 	}
-	exit(0);
+	return 0;
 		]])],
 		[
 			AC_MSG_RESULT([yes])
@@ -2066,7 +2066,7 @@
 	err = getaddrinfo(name, TEST_PORT, &hints, &gai_ai);
 	if (err != 0) {
 		fprintf(stderr, "getaddrinfo failed (%s)", gai_strerror(err));
-		exit(1);
+		return 1;
 	}
 
 	for (ai = gai_ai; ai != NULL; ai = ai->ai_next) {
@@ -2079,10 +2079,10 @@
 
 		if (ai->ai_family == AF_INET && err != 0) {
 			perror("getnameinfo");
-			exit(2);
+			return 2;
 		}
 	}
-	exit(0);
+	return 0;
 		]])],
 		[
 			AC_MSG_RESULT([yes])
@@ -2103,7 +2103,7 @@
 if test "x$check_for_conflicting_getspnam" = "x1"; then
 	AC_MSG_CHECKING([for conflicting getspnam in shadow.h])
 	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[ #include <shadow.h> ]],
-		[[ exit(0); ]])],
+		[[ return 0; ]])],
 		[
 			AC_MSG_RESULT([no])
 		],
@@ -2191,12 +2191,12 @@
 
 	fd = fopen(DATA,"w");
 	if(fd == NULL)
-		exit(1);
+		return 1;
 
 	if ((rc = fprintf(fd ,"%x (%s)\n", OPENSSL_VERSION_NUMBER, OPENSSL_VERSION_TEXT)) <0)
-		exit(1);
+		return 1;
 
-	exit(0);
+	return 0;
 	]])],
 	[
 		ssl_header_ver=`cat conftest.sslincver`
@@ -2226,12 +2226,12 @@
 
 	fd = fopen(DATA,"w");
 	if(fd == NULL)
-		exit(1);
+		return 1;
 
 	if ((rc = fprintf(fd ,"%x (%s)\n", SSLeay(), SSLeay_version(SSLEAY_VERSION))) <0)
-		exit(1);
+		return 1;
 
-	exit(0);
+	return 0;
 	]])],
 	[
 		ssl_library_ver=`cat conftest.ssllibver`
@@ -2261,7 +2261,7 @@
 #include <string.h>
 #include <openssl/opensslv.h>
 	]], [[
-	exit(SSLeay() == OPENSSL_VERSION_NUMBER ? 0 : 1);
+	return (SSLeay() == OPENSSL_VERSION_NUMBER ? 0 : 1);
 	]])],
 	[
 		AC_MSG_RESULT([yes])
@@ -2339,7 +2339,7 @@
 #include <string.h>
 #include <openssl/evp.h>
 	]], [[
-	exit(EVP_aes_192_cbc() == NULL || EVP_aes_256_cbc() == NULL);
+	return (EVP_aes_192_cbc() == NULL || EVP_aes_256_cbc() == NULL);
 	]])],
 	[
 		AC_MSG_RESULT([no])
@@ -2358,7 +2358,7 @@
 #include <string.h>
 #include <openssl/evp.h>
 	]], [[
-	exit(EVP_aes_128_ctr() == NULL ||
+	return (EVP_aes_128_ctr() == NULL ||
 	    EVP_aes_192_cbc() == NULL ||
 	    EVP_aes_256_cbc() == NULL);
 	]])],
@@ -2379,7 +2379,7 @@
 #include <string.h>
 #include <openssl/evp.h>
 	]], [[
-	exit(EVP_aes_128_gcm() == NULL ||
+	return (EVP_aes_128_gcm() == NULL ||
 	    EVP_aes_256_gcm() == NULL ||
 	    EVP_CTRL_GCM_SET_IV_FIXED == 0 ||
 	    EVP_CTRL_GCM_IV_GEN == 0 ||
@@ -2410,7 +2410,7 @@
 #include <openssl/evp.h>
 	]], [[
 	if(EVP_DigestUpdate(NULL, NULL,0))
-		exit(0);
+		return 0;
 	]])],
 	[
 		AC_MSG_RESULT([yes])
@@ -2505,7 +2505,7 @@
 #include <string.h>
 #include <openssl/rand.h>
 	]], [[
-	exit(RAND_status() == 1 ? 0 : 1);
+	return (RAND_status() == 1 ? 0 : 1);
 	]])],
 	[
 		OPENSSL_SEEDS_ITSELF=yes
@@ -2702,7 +2702,7 @@
 	[[ int i = $seccomp_audit_arch;
 	   errno = 0;
 	   prctl(PR_SET_SECCOMP, SECCOMP_MODE_FILTER, NULL, 0, 0);
-	   exit(errno == EFAULT ? 0 : 1); ]])],
+	   return (errno == EFAULT ? 0 : 1); ]])],
 	[ AC_MSG_RESULT([yes]) ], [
 		AC_MSG_RESULT([no])
 		# Disable seccomp filter as a target
@@ -2756,7 +2756,7 @@
 	tv.tv_sec = 1;
 	tv.tv_usec = 0;
 	r = select(fd+1, &fds, NULL, NULL, &tv);
-	exit (r == -1 ? 1 : 0);
+	return (r == -1 ? 1 : 0);
 	]])],
 	[AC_MSG_RESULT([yes])
 	 select_works_with_rlimit=yes],
@@ -2782,7 +2782,7 @@
 
 	rl_zero.rlim_cur = rl_zero.rlim_max = 0;
 	r = setrlimit(RLIMIT_NOFILE, &rl_zero);
-	exit (r == -1 ? 1 : 0);
+	return (r == -1 ? 1 : 0);
 	]])],
 	[AC_MSG_RESULT([yes])
 	 rlimit_nofile_zero_works=yes],
@@ -2801,7 +2801,7 @@
 		struct rlimit rl_zero;
 
 		rl_zero.rlim_cur = rl_zero.rlim_max = 0;
-		exit(setrlimit(RLIMIT_FSIZE, &rl_zero) != 0);
+		return (setrlimit(RLIMIT_FSIZE, &rl_zero) != 0);
 	]])],
 	[AC_MSG_RESULT([yes])],
 	[AC_MSG_RESULT([no])
@@ -2925,7 +2925,7 @@
 	long long i, llmin, llmax = 0;
 
 	if((f = fopen(DATA,"w")) == NULL)
-		exit(1);
+		return 1;
 
 #if defined(LLONG_MIN) && defined(LLONG_MAX)
 	fprintf(stderr, "Using system header for LLONG_MIN and LLONG_MAX\n");
@@ -2944,16 +2944,16 @@
 	    || llmax - 1 > llmax || llmin == llmax || llmin == 0
 	    || llmax == 0 || llmax < LONG_MAX || llmin > LONG_MIN) {
 		fprintf(f, "unknown unknown\n");
-		exit(2);
+		return 2;
 	}
 
 	if (fprint_ll(f, llmin) < 0)
-		exit(3);
+		return 3;
 	if (fprint_ll(f, llmax) < 0)
-		exit(4);
+		return 4;
 	if (fclose(f) < 0)
-		exit(5);
-	exit(0);
+		return 5;
+	return 0;
 		]])],
 		[
 			llong_min=`$AWK '{print $1}' conftest.llminmax`
@@ -3337,11 +3337,11 @@
 	strcpy(expected_out, "9223372036854775807");
 	snprintf(buf, mazsize, "%lld", num);
 	if(strcmp(buf, expected_out) != 0)
-		exit(1);
-	exit(0);
+		return 1;
+	return 0;
 }
 #else
-main() { exit(0); }
+main() { return 0; }
 #endif
 		]])], [ true ], [ AC_DEFINE([BROKEN_SNPRINTF]) ],
 		AC_MSG_WARN([cross compiling: Assuming working snprintf()])
@@ -3425,11 +3425,11 @@
 		]], [[
 #ifdef msg_accrights
 #error "msg_accrights is a macro"
-exit(1);
+return 1;
 #endif
 struct msghdr m;
 m.msg_accrights = 0;
-exit(0);
+return 0;
 		]])],
 		[ ac_cv_have_accrights_in_msghdr="yes" ],
 		[ ac_cv_have_accrights_in_msghdr="no" ]
@@ -3486,11 +3486,11 @@
 		]], [[
 #ifdef msg_control
 #error "msg_control is a macro"
-exit(1);
+return 1;
 #endif
 struct msghdr m;
 m.msg_control = 0;
-exit(0);
+return 0;
 		]])],
 		[ ac_cv_have_control_in_msghdr="yes" ],
 		[ ac_cv_have_control_in_msghdr="no" ]
@@ -3912,22 +3912,22 @@
 
 	fd = fopen(DATA,"w");
 	if(fd == NULL)
-		exit(1);
+		return 1;
 
 #if defined (_PATH_MAILDIR)
 	if ((rc = fprintf(fd ,"_PATH_MAILDIR:%s\n", _PATH_MAILDIR)) <0)
-		exit(1);
+		return 1;
 #elif defined (MAILDIR)
 	if ((rc = fprintf(fd ,"MAILDIR:%s\n", MAILDIR)) <0)
-		exit(1);
+		return 1;
 #elif defined (_PATH_MAIL)
 	if ((rc = fprintf(fd ,"_PATH_MAIL:%s\n", _PATH_MAIL)) <0)
-		exit(1);
+		return 1;
 #else
-	exit (2);
+	return 2;
 #endif
 
-	exit(0);
+	return 0;
 		]])],
 		[
 	 	    maildir_what=`awk -F: '{print $1}' conftest.maildir`
@@ -4162,12 +4162,12 @@
 
 	fd = fopen(DATA,"w");
 	if(fd == NULL)
-		exit(1);
+		return 1;
 
 	if ((rc = fprintf(fd,"%s", _PATH_STDPATH)) < 0)
-		exit(1);
+		return 1;
 
-	exit(0);
+	return 0;
 		]])],
 		[ user_path=`cat conftest.stdpath` ],
 		[ user_path="/usr/bin:/bin:/usr/sbin:/sbin" ],
