diff -ruN dhcdbd-2.5.orig/include/dhcdbd.h dhcdbd-2.5/include/dhcdbd.h
--- dhcdbd-2.5.orig/include/dhcdbd.h	2007-02-15 12:39:30.000000000 -0600
+++ dhcdbd-2.5/include/dhcdbd.h	2007-02-15 15:07:23.000000000 -0600
@@ -64,12 +64,12 @@
 #endif
 
 #ifndef DHCDBD_DHCLIENT_CONF_PFX
-#define DHCDBD_DHCLIENT_CONF_PFX "/etc/dhclient"
+#define DHCDBD_DHCLIENT_CONF_PFX "/etc/dhcp/dhclient"
 #endif
 
-#ifndef DHCDBD_DHCLIENT_CONF_SEP
+/*#ifndef DHCDBD_DHCLIENT_CONF_SEP*/
 #define DHCDBD_DHCLIENT_CONF_SEP "-"
-#endif
+/*#endif*/
 
 #ifndef DHCDBD_DHCLIENT_CONF_SFX
 #define DHCDBD_DHCLIENT_CONF_SFX ".conf"
diff -ruN dhcdbd-2.5.orig/src/dbus_service.c dhcdbd-2.5/src/dbus_service.c
--- dhcdbd-2.5.orig/src/dbus_service.c	2007-02-15 12:39:30.000000000 -0600
+++ dhcdbd-2.5/src/dbus_service.c	2007-02-15 12:44:43.000000000 -0600
@@ -1216,7 +1216,7 @@
     return (cs);
 
 give_up:
-    dbus_connection_close(connection);
+    dbus_connection_unref(connection);
     dbus_shutdown();
 
     return (0L);
@@ -1250,7 +1250,7 @@
     tdestroy(cs->roots, free_root);
     cs->roots = 0L;
 
-    dbus_connection_close(cs->connection);
+    dbus_connection_unref(cs->connection);
     dbus_shutdown();
     free(cs);
 }
diff -ruN dhcdbd-2.5.orig/src/dhcdbd.c dhcdbd-2.5/src/dhcdbd.c
--- dhcdbd-2.5.orig/src/dhcdbd.c	2007-02-15 12:39:30.000000000 -0600
+++ dhcdbd-2.5/src/dhcdbd.c	2007-02-15 15:06:11.000000000 -0600
@@ -833,9 +833,8 @@
                 ) {
                 dhcdbd_debug ("Running dhclient for IF_UP %s", d_if->name);
 
-                snprintf (dhc_if_conf_file, 1024, "%s%s%s%s",
-                                  DHCDBD_DHCLIENT_CONF_PFX, DHCDBD_DHCLIENT_CONF_SEP,
-                                  d_if->name, DHCDBD_DHCLIENT_CONF_SFX);
+                snprintf (dhc_if_conf_file, 1024, "%s%s",
+                                  DHCDBD_DHCLIENT_CONF_PFX, DHCDBD_DHCLIENT_CONF_SFX);
 
                 if (access (dhc_if_conf_file, R_OK) == 0)
                         n_args += 2;
Files dhcdbd-2.5.orig/src/.dhcdbd.c.swp and dhcdbd-2.5/src/.dhcdbd.c.swp differ
