--- a/Python-3.3.3/Modules/posixmodule.c
+++ b/Python-3.3.3/Modules/posixmodule.c
@@ -114,7 +114,7 @@
 #undef HAVE_SCHED_SETAFFINITY
 #endif
 
-#if defined(HAVE_SYS_XATTR_H) && defined(__GLIBC__)
+#if defined(HAVE_SYS_XATTR_H)
 #define USE_XATTRS
 #endif
 
@@ -122,6 +122,10 @@
 #include <sys/xattr.h>
 #endif
 
+#if defined(__linux__)
+#include <linux/limits.h>
+#endif
+
 #if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__APPLE__)
 #ifdef HAVE_SYS_SOCKET_H
 #include <sys/socket.h>
