--- a/Python-3.3.3/Modules/posixmodule.c
+++ b/Python-3.3.3/Modules/posixmodule.c
@@ -2837,8 +2837,13 @@
     else
 #endif
 #ifdef HAVE_LCHMOD
-    if ((!follow_symlinks) && (dir_fd == DEFAULT_DIR_FD))
+    if ((!follow_symlinks) && (dir_fd == DEFAULT_DIR_FD)) {
         result = lchmod(path.narrow, mode);
+        fchmodat_nofollow_unsupported =
+                         result &&
+                         ((errno == ENOTSUP) || (errno == EOPNOTSUPP)) &&
+                         !follow_symlinks;
+    }
     else
 #endif
 #ifdef HAVE_FCHMODAT
