diff --git a/include/extern.h b/include/extern.h
index f00e4cf..b70a801 100644
--- a/include/extern.h
+++ b/include/extern.h
@@ -5,6 +5,8 @@
 #include <signal.h>
 #include <time.h>
 #include <netinet/in.h>
+#include <sys/types.h>
+#include <asm-generic/param.h>
 
 #include "logmessage.h"
 #include "xmalloc.h"
diff --git a/include/sundries.h b/include/sundries.h
index 4379982..55063ad 100644
--- a/include/sundries.h
+++ b/include/sundries.h
@@ -10,7 +10,7 @@
 #include <stdarg.h>
 #include <stdlib.h>
 #if !defined(bool_t) && !defined(__GLIBC__)
-#include <rpc/types.h>
+//#include <rpc/types.h>
 #endif
 
 extern int mount_mount_quiet;
diff --git a/src/ifdown.c b/src/ifdown.c
index cef62af..747d3da 100644
--- a/src/ifdown.c
+++ b/src/ifdown.c
@@ -20,7 +20,7 @@ char *v_ifdown = "@(#)ifdown.c  1.10  21-Apr-1997  miquels@cistron.nl";
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/time.h>
-#include <sys/errno.h>
+#include <errno.h>
 
 #include <net/if.h>
 #include <netinet/in.h>
diff --git a/src/pidfile.c b/src/pidfile.c
index 987b8a0..6d57962 100644
--- a/src/pidfile.c
+++ b/src/pidfile.c
@@ -7,7 +7,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
-#include <wait.h>
+#include <sys/wait.h>
 
 #include "extern.h"
 #include "watch_err.h"
