--- apps/mplayer.te	2010-12-13 15:11:01.000000000 +0100
+++ ../../../refpolicy/policy/modules/apps/mplayer.te	2011-01-07 21:03:00.230000033 +0100
@@ -197,6 +197,7 @@
 dev_rwx_zero(mplayer_t)
 # Access to DVD/CD/V4L
 dev_read_video_dev(mplayer_t)
+dev_write_video_dev(mplayer_t)
 # Audio, alsa.conf
 dev_read_sound_mixer(mplayer_t)
 dev_write_sound_mixer(mplayer_t)
@@ -226,6 +227,9 @@
 miscfiles_read_fonts(mplayer_t)
 
 userdom_use_user_terminals(mplayer_t)
+domain_use_interactive_fds(mplayer_t)
+
+#
 # Read media files
 userdom_list_user_tmp(mplayer_t)
 userdom_read_user_tmp_files(mplayer_t)
