diff -Naur util-linux-2.25.orig/sys-utils/switch_root.c util-linux-2.25/sys-utils/switch_root.c
--- util-linux-2.25.orig/sys-utils/switch_root.c	2014-06-25 09:35:44.333812673 +0000
+++ util-linux-2.25/sys-utils/switch_root.c	2014-08-12 14:54:40.752012106 +0000
@@ -25,6 +25,7 @@
 #include <sys/stat.h>
 #include <sys/statfs.h>
 #include <sys/param.h>
+#include <sys/reg.h>
 #include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -47,6 +48,16 @@
 #define MNT_DETACH       0x00000002	/* Just detach from the tree */
 #endif
 
+#ifndef __SWORD_TYPE
+#if __WORDSIZE == 32
+# define __SWORD_TYPE           int
+#elif __WORDSIZE == 64
+# define __SWORD_TYPE           long int
+#else
+# error
+#endif
+#endif
+
 /* remove all files/directories below dirName -- don't cross mountpoints */
 static int recursiveRemove(int fd)
 {
