--- glibc-2.6.1/elf/rtld.c.mps	2008-05-16 15:55:06.000000000 +0200
+++ glibc-2.6.1/elf/rtld.c	2008-05-16 16:18:21.000000000 +0200
@@ -90,8 +90,7 @@ INTDEF(_dl_argv)
 uintptr_t __stack_chk_guard attribute_relro;
 #endif
 #ifdef ENABLE_OLD_SSP_COMPAT
-uintptr_t __guard_local attribute_relro attribute_hidden __attribute__ ((nocommon));
-weak_alias (__guard_local, __guard);
+uintptr_t __guard attribute_relro;
 #endif
 
 /* Only exported for architectures that don't store the pointer guard
@@ -1822,7 +1821,7 @@ ERROR: ld.so: object '%s' cannot be load
   /* Set up the stack checker's canary.  */
   uintptr_t stack_chk_guard = _dl_setup_stack_chk_guard ();
 #ifdef ENABLE_OLD_SSP_COMPAT
-  __guard_local = stack_chk_guard;
+  __guard = stack_chk_guard;
 #endif
 #ifdef THREAD_SET_STACK_GUARD
   THREAD_SET_STACK_GUARD (stack_chk_guard);
--- glibc-2.6.1/csu/libc-start.c.mps	2008-05-16 15:54:38.000000000 +0200
+++ glibc-2.6.1/csu/libc-start.c	2008-05-16 15:54:54.000000000 +0200
@@ -38,8 +38,7 @@ uintptr_t __stack_chk_guard attribute_re
 # endif
 #endif
 #ifdef ENABLE_OLD_SSP_COMPAT
-uintptr_t __guard_local attribute_relro attribute_hidden __attribute__ ((nocommon,weak));
-weak_alias (__guard_local, __guard);
+uintptr_t __guard attribute_relro;
 #endif
 
 #ifdef HAVE_PTR_NTHREADS
@@ -148,7 +147,7 @@ LIBC_START_MAIN (int (*main) (int, char 
   /* Set up the stack checker's canary.  */
   uintptr_t stack_chk_guard = _dl_setup_stack_chk_guard ();
 #ifdef ENABLE_OLD_SSP_COMPAT
-  __guard_local = stack_chk_guard;
+  __guard = stack_chk_guard;
 #endif
 #  ifdef THREAD_SET_STACK_GUARD
   THREAD_SET_STACK_GUARD (stack_chk_guard);

